### DISCUSSION TITLE
Video quality from the player very expired

### DISCUSSION THREAD

**Discussion by annibal1** [OvenMediaLabs/OvenMediaEngine#1196] [Help]
I have installed OngiMediaengine and Overplayer I have set the OBS parameters as per attached pictures But the streaming quality is very poor Do you have any ideas how to improve the video quality? Thank you
![Immagine 2023-04-23 142431](https://user-images.githubusercontent.com/50693441/233904127-8b6d6b04-f5c0-4b54-8f2b-f753de312c64.png)
ie

**Comment by getroot** [comment]
We need your ovenmediaengine.log file and Server.xml file to accurately troubleshoot.

If you are using OME to encode video, you should check if there are enough cpu resources. And check the network bandwidth.

**Comment by annibal1** [comment]
attached my Server.xml file
I noticed the log file is not updated
I installed the non Docker version
and in the /usr/share/ovenmediaengine/logs folder
but it is not updated
maybe it's somewhere else?
Thank you for your answer

[Server.xml.txt](https://github.com/AirenSoft/OvenMediaEngine/files/11307280/Server.xml.txt)

**Reply by naanlizard** [reply]
Please also put a screenshot of your stream layout and video settings tab in obs

Probably the first thing I'd suggest trying is set your CPU preset to very fast and keyframe interval to 2, or 4. I'll explain in more detail later

**Reply by getroot** [reply]
https://airensoft.gitbook.io/ovenmediaengine/logs-and-statistics

The log files are located in /var/log/ovenmediaengine/.

**Comment by getroot** [comment]
What exactly does "very expired" and "very poor" mean? Are there stuttering in playback? Or is it that the image is being squashed?

**Comment by annibal1** [comment]
attached the log file
refer to yesterday's data and not today's data that I tried from work and I have no bandwidth
as for the quality I mean that it looks grainy
when i tried it yesterday i tried it with a football game and the video was very grainy especially with moving balls
I don't know if I was clear enough :(
[ovenmediaengine.log](https://github.com/AirenSoft/OvenMediaEngine/files/11308116/ovenmediaengine.log)

**Reply by getroot** [reply]
I thought it was a stuttering problem, but the image is grainy. It would be good to take the advice of @naanlizard above.

**Comment by annibal1** [comment]
follow the advice of [Naanlizard](https://github.com/naanlizard)??
from obs??

**Reply by getroot** [reply]
https://github.com/AirenSoft/OvenMediaEngine/discussions/1196#discussioncomment-5704579

**Comment by annibal1** [comment]
I did other tests by increasing the parameters as from attached images
I am testing on a football match and when I see the ball go from one side to the other it seems that it goes to Slowness
![Immagine 2023-04-24 185443](https://user-images.githubusercontent.com/50693441/234064854-86f2195b-f637-4d5c-974d-25dfccdeea9c.png)
![Immagine 2023-04-24 185418](https://user-images.githubusercontent.com/50693441/234064859-c8c6c335-7d1b-42a4-af19-dd77f285fa74.png)

**Reply by getroot** [reply]
How is the quality when played back in LLHLS?
If the quality is the same in LLHLS, the best solution would be to fine-tune the encoding quality in OBS until you are satisfied. It should be noted that since WebRTC does not support b-frames, it must be disabled unconditionally.
If the quality is what you want in LLHLS, the jitter between OBS and OME may be the cause. This may help with Enable network optimizations in OBS.

![image](https://user-images.githubusercontent.com/17779085/234165424-b4c3fc7a-fa14-4e4c-aef5-ba8a9b210113.png)

**Reply by getroot** [reply]
OBS's x264 may not output the quality you want. Trying out the NVENC hardware encoder is also a good idea.

![image](https://user-images.githubusercontent.com/17779085/234165626-f0a772d8-5bff-4cae-88d7-5c6d96b0614d.png)

**Comment by annibal1** [comment]
I'm trying to see the flow with this link
https: // my_domain: 3334/app/stream/llhls.m3u8 but downloads me a file and VLC does not allow playback it gives me error

**Reply by getroot** [reply]
Try putting LLHLS URL in OvenPlayer like WebRTC. It should work fine in VLC as well. (Of course VLC doesn't support LLHLS, so it plays in Legacy HLS.)

**Comment by annibal1** [comment]
I waited a few days before responding because I carried out various tests
It seems that the problem of quality is now better following your advice but gni shots so even with VLC
is there any other configuration that I can follow ??
Change some confidents in the server ??
I would not know

**Reply by naanlizard** [reply]
> but gni shots so even with VLC

Can you clarify this for me please?
